config/yaml_test.go

```diff
@@ -255,6 +255,17 @@ func TestTypeOfTypes(t *testing.T) {
 		tts := typeStructStruct{}
 		err := provider.Get(Root).PopulateStruct(&tts)
 		assert.NoError(t, err)
+		assert.Equal(t, userDefinedTypeInt(123), tts.TypeStruct.TestInt)
+		assert.Equal(t, userDefinedTypeUInt(456), tts.TypeStruct.TestUInt)
+		assert.Equal(t, userDefinedTypeFloat(123.456), tts.TypeStruct.TestFloat)
```

Comment:
I remember we had some issues with converting a string that was written as int, but the target type is actually a float, could you test it here too?